wcfsetup/install/files/lib/data/comment/CommentAction.class.php

```diff
@@ -501,37 +501,28 @@ public function triggerPublication()
             }
 
             // fire notification event
-            if (UserNotificationHandler::getInstance()->getObjectTypeID($objectType->objectType . '.notification')) {
+            if (UserNotificationHandler::getInstance()->getEvent($objectType->objectType . '.notification', 'comment')) {
                 $notificationObject = new CommentUserNotificationObject($comment->getDecoratedObject());
                 $notificationObjectType = UserNotificationHandler::getInstance()->getObjectTypeProcessor($objectType->objectType . '.notification');
+                \assert($notificationObjectType instanceof ICommentUserNotificationObjectType);
 
                 if ($notificationObjectType instanceof IMultiRecipientCommentUserNotificationObjectType) {
                     $recipientIDs = $notificationObjectType->getRecipientIDs($comment->getDecoratedObject());
-
-                    // make sure that the comment's author gets no notification
-                    $recipientIDs = \array_diff($recipientIDs, [$comment->getUserID()]);
-
-                    if (!empty($recipientIDs)) {
-                        UserNotificationHandler::getInstance()->fireEvent(
-                            'comment',
-                            $objectType->objectType . '.notification',
-                            $notificationObject,
-                            $recipientIDs
-                        );
-                    }
                 } else {
-                    /** @var ICommentUserNotificationObjectType $notificationObjectType */
-                    $userID = $notificationObjectType->getOwnerID($comment->commentID);
-                    if ($userID != $comment->getUserID()) {
-                        UserNotificationHandler::getInstance()->fireEvent(
-                            'comment',
-                            $objectType->objectType . '.notification',
-                            $notificationObject,
-                            [$userID],
-                            ['objectUserID' => $userID]
-                        );
-                    }
+                    $recipientIDs = [];
                 }
+
+                $recipientIDs[] = $notificationObjectType->getOwnerID($comment->commentID);
```

Comment:
This can legitimately return a `0`. Maybe its wise to filter the `$recipientIDs` parameter in `UserNotificationHandler::fireEvent()` and simply return early if there are no recipient?